### ISSUE TITLE
delete crash

### ISSUE THREAD

**Issue by fluxionary** [mt-mods/mail#97]
using the most recent version (https://github.com/mt-mods/mail/commit/fab6f1a1e5808a4e2821155859b78c7829e36ddf)

```
2023-06-04 19:03:45: ERROR[Main]: ServerError: AsyncErr: Lua: Runtime error from mod 'mail' in callback on_playerReceiveFields(): .../mt/5.7.0/Minetest_test/bin/../mods/mail_mod/storage.lua:119: attempt to index a nil value
2023-06-04 19:03:45: ERROR[Main]: stack traceback:
2023-06-04 19:03:45: ERROR[Main]: 	.../mt/5.7.0/Minetest_test/bin/../mods/mail_mod/storage.lua:119: in function 'delete_mail'
2023-06-04 19:03:45: ERROR[Main]: 	...t/5.7.0/Minetest_test/bin/../mods/mail_mod/ui/events.lua:190: in function 'func'
2023-06-04 19:03:45: ERROR[Main]: 	...inetest_test/bin/../builtin/profiler/instrumentation.lua:108: in function <...inetest_test/bin/../builtin/profiler/instrumentation.lua:101>
2023-06-04 19:03:45: ERROR[Main]: 	.../mt/5.7.0/Minetest_test/bin/../builtin/game/register.lua:446: in function <.../mt/5.7.0/Minetest_test/bin/../builtin/game/register.lua:432>
```

the user reports:

* What I was trying to do is to delete all mails. "Select all" confuses me (do selected mails have darker or lighter color?), so I just selected them all with shift+arrows, then pressed "delete" and nothing happened. After that I pressed "Select all" again and pressed "delete" again and it crashed.

**Comment by BuckarooBanzay** [comment]
Hm, this looks like the player doesn't have mails in the inbox to iterate through:
https://github.com/mt-mods/mail/blob/fab6f1a1e5808a4e2821155859b78c7829e36ddf/storage.lua#L119
:confused: 
(or maybe they are deleted already...)

**Comment by Athozus** [comment]
@BuckarooBanzay you're removing the multiple deletion system by adding a break. I just added a verification of entry lengths, and it is fixed.

**Comment by S-S-X** [comment]
> @S-S-X I do not think it does anything, when you have a 0 length it does a weird thing.

Just try it yourself, it is basic Lua and nothing else. Loop block is either executed or not and if table is empty it is not executed, `if #t > 0 then` is completely unnecessary because `for i=#t,1,-1 do` already does exactly same.

If you do not believe your eyes then just try it. Reason for crash is in inner loop which attempts to index past outer loop table after first removal.

**Comment by BuckarooBanzay** [comment]
should be fixed with #99 otherwise reopen please
